lib/src/global.dart

```diff
@@ -68,11 +68,11 @@ Future<dynamic> deleteOfflineRegion(int id, {String accessToken}) =>
       },
     );
 
-Future<dynamic> downloadOfflineRegion(
-  OfflineRegion region, {
+Future<OfflineRegion> downloadOfflineRegion(
+  OfflineRegionDefinition region, {
```

Comment:
This breaks the api for android as well. However if you go down the base class route, dart should just cast it here, and existing implementations should not break.

This might also cause issues in json.encode(region._toJson()) for android, as this code probably expects the id?

Reply:
> This breaks the api for android as well. However if you go down the base class route, dart should just cast it here, and existing implementations should not break.

True, that's a good reason to go with inheritance.

> This might also cause issues in json.encode(region._toJson()) for android, as this code probably expects the id?

I don't think so, the Android PR mentions "Do not respect id passed into downloadOfflineRegion via OfflineRegionData, as it will be assigned by createOfflineRegion." The changes are in `fromOfflineRegion`:
https://github.com/tobrun/flutter-mapbox-gl/pull/491/files#diff-dbcc20feb45d56319fdc09f95c724ae0202964e5befc014f5009981ad449929eR85-R101

Reply:
> > This breaks the api for android as well. However if you go down the base class route, dart should just cast it here, and existing implementations should not break.
> 
> True, that's a good reason to go with inheritance.

This is a kuch cleaner api and it might be worth that from the beginning rather than be burdened with unnecessary cruft from a feature that is not even functional in the latest release. I should be able to make the corresponding changes on Android in the next couple of days.

Reply:
Just saw the other discussion and change re inheritance and this makes sense now.